Hantar/code.js

```diff
@@ -0,0 +1,403 @@
+        }
+        else
+        {
+            range_parts = items[i].split("-");
+            list = list.concat(range(range_parts[0] * 1, range_parts[1] * 1));
+        }
+    }
+    return list;
+}
+
+function gen_list_signature(num_list)
+{
+    if (num_list.length == 0)
+    {
+        return "";
+    }
+    
+    var item = num_list[0];
+    var prev_item;
+    var list_signature = num_list[0];
+    var range_count = 1;
+    for (var i = 1, len = num_list.length; i < len; i++)
+    {
+        prev_item = item;
+        item = num_list[i];
+        if (item == prev_item + 1)
+        {
+            range_count++;
+            if (i == len - 1)
+            {
+                list_signature += "-" + item;
+            }
+        }
+        else
+        {
+            if (range_count > 1)
+            {
+                list_signature += "-" + prev_item;
+            }
+            list_signature += "," + item;
+            range_count = 1;
+        }
+    }
+    return list_signature;
+}
+
+function save_num_lists(num_lists)
+{
+    save(LS_KEY_UNMARKED_LIST, gen_list_signature(num_lists[0]));
+    save(LS_KEY_MARKED_LIST, gen_list_signature(num_lists[1]));
+}
+
+function prev_session_exists()
+{
+    return load(LS_KEY_SESSION_ID) != null
+}
+
+function get_session_id()
+{
+    return load(LS_KEY_SESSION_ID);
+}
+
+function get_session_num_lists()
+{
+    return [
+        gen_num_list(load(LS_KEY_UNMARKED_LIST)),
+        gen_num_list(load(LS_KEY_MARKED_LIST))
+    ];
+}
+
+function add_items_to_lists(elm_list, data_lists)
+{
+    var node;
+    for (var i = 0; i < elm_list.length; i++)
+    {
+        for (var j = 0; j < data_lists[i].length; j++)
+        {
+            node = document.createElement("li");
+            node.setAttribute("value", data_lists[i][j]);
+            node.innerHTML = data_lists[i][j];
+            elm_list[i].appendChild(node);
+        }
+    }
+}
+
+function get_list_elms()
+{
+    elm_lists = [];
+    for (var i = 0, len = ELM_ID_LISTS.length; i < len; i++)
+    {
+        elm_lists.push(document.querySelector("#" + ELM_ID_LISTS[i]));
+    }
+    return elm_lists;
+}
+
+function cont_prev_session()
+{
+    log("CONTINUE PREVIOUS SESSION - Checking previous session...");
+    if (prev_session_exists())
+    {
+        log("Connecting to previous session (" + get_session_id() + ")...");
+        init_main_screen();
+    }
+    else
+    {
+        log("No previous session.");
+        alert("There is no existing session!");
+    }
+}
+
+function new_session()
+{
+    log("CREATE NEW SESSION - Checking previous session...");
+    if (!prev_session_exists() || confirm("Are you sure? This will delete all previous data!"))
+    {
+        log("Starting new session...");
+        var session_id = new Date().getTime();
+        save(LS_KEY_SESSION_ID, session_id);
+        log("New session id: " + session_id);
+        var sig;
+        do
+        {
+            sig = prompt("Please enter a valid number list:");
```

Comment:
`prompt` is almost as ugly as `alert`. Also the input your user is asked to enter is requested in a very indicative way. Create a better UI.